app/services/MarriageAllowanceService.scala

```diff
@@ -45,7 +45,7 @@ class MarriageAllowanceService @Inject()(dataConnector: MarriageAllowanceDESConn
   val startTaxYear: Int = appConfig.START_TAX_YEAR
   val maSupportedYearsCount: Int = appConfig.MA_SUPPORTED_YEARS_COUNT
 
-  def currentTaxYear: Int = TaxYear.current.startYear
+  val currentTaxYear: Int = appConfig.currentTaxYear()
```

Comment:
This could be it. A val will be set the first time it is ever called. Where a def re evaluates